sources/sources.go

```diff
@@ -3,163 +3,116 @@ package sources
 	"fmt"
-	"github.com/KYVENetwork/ksync/collectors/pool"
+	"github.com/KYVENetwork/ksync/sources/helpers"
 	"github.com/KYVENetwork/ksync/types"
 	"github.com/KYVENetwork/ksync/utils"
 	"gopkg.in/yaml.v2"
 	"io/ioutil"
 	"net/http"
 	"strconv"
-	"strings"
-)
-
-const (
-	redNo    = "\033[31m" + "NO" + "\033[0m"
-	greenYes = "\033[32m" + "YES" + "\033[0m"
 )
 
 func FormatOutput(entry *types.Entry, chainId string) (string, string, string) {
 	var blockKey, stateKey, heightKey string
 	if chainId == utils.ChainIdMainnet {
-		blockKey, stateKey, heightKey = formatKeys(entry.Kyve.BlockStartKey, entry.Kyve.LatestBlockKey, entry.Kyve.StateStartKey, entry.Kyve.LatestStateKey)
+		blockKey, stateKey, heightKey = helpers.FormatKeys(entry.Kyve.BlockStartKey, entry.Kyve.LatestBlockKey, entry.Kyve.StateStartKey, entry.Kyve.LatestStateKey)
 	} else if chainId == utils.ChainIdKaon {
-		blockKey, stateKey, heightKey = formatKeys(entry.Kaon.BlockStartKey, entry.Kaon.LatestBlockKey, entry.Kaon.StateStartKey, entry.Kaon.LatestStateKey)
+		blockKey, stateKey, heightKey = helpers.FormatKeys(entry.Kaon.BlockStartKey, entry.Kaon.LatestBlockKey, entry.Kaon.StateStartKey, entry.Kaon.LatestStateKey)
 	}
 	return blockKey, stateKey, heightKey
 }
 
-func GetSourceRegistry(url string) (*types.SourceRegistry, error) {
-	response, err := http.Get(url)
-	if err != nil {
-		return nil, err
-	}
-
-	if response.StatusCode != 200 {
-		return nil, fmt.Errorf("got status code %d != 200", response.StatusCode)
-	}
-
-	data, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	var sourceRegistry types.SourceRegistry
-
-	err = yaml.Unmarshal(data, &sourceRegistry)
-	if err != nil {
-		return nil, err
+func GetPoolIds(chainId, source, blockPoolId, snapshotPoolId, registryUrl string, blockPoolRequired, snapshotPoolRequired bool) (int64, int64, error) {
+	if source == "" && blockPoolId == "" && blockPoolRequired {
+		return 0, 0, fmt.Errorf("either --source or --block-pool-id are required")
 	}
-
-	r, err := loadLatestPoolData(sourceRegistry)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load latest pool data: %v", err)
+	if source == "" && snapshotPoolId == "" && snapshotPoolRequired {
+		return 0, 0, fmt.Errorf("either --source or --snapshot-pool-id are required")
 	}
 
-	return r, nil
-}
+	var bId, sId int64
 
-func loadLatestPoolData(sourceRegistry types.SourceRegistry) (*types.SourceRegistry, error) {
-	for _, entry := range sourceRegistry.Entries {
-		if entry.Kyve.BlockPoolID != nil {
-			poolResponse, err := pool.GetPoolInfo(utils.RestEndpointMainnet, int64(*entry.Kyve.BlockPoolID))
-			if err != nil {
-				return nil, err
-			}
-			entry.Kyve.BlockStartKey = &poolResponse.Pool.Data.StartKey
-			entry.Kyve.LatestBlockKey = &poolResponse.Pool.Data.CurrentKey
+	if source != "" {
+		bIdRaw, sIdRaw, err := getPoolsBySource(chainId, source, registryUrl)
+		if err != nil {
+			return 0, 0, fmt.Errorf("failed to load poolIds for source: %s", err)
```

Comment:
better error message would be `fmt.Errorf("failed to load pool Ids for source %s from %s: %w", source, registryUrl, err)`